### PR TITLE
Api test cases

### PR DESCRIPTION
**Pull request by jwoody02**
added 4 or 5 test cases #288

### CODE REVIEW

File: interactive/dashboard/api.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def post(self, request):
         if not request.session.session_key:
             request.session.create()
         logging.debug("saving model options for user: " + request.session.session_key)
-        logging.debug("received options: " + str(request.body))
+        print("received options: ",request.body)
```
Comment (K20shores) [inline interactive/dashboard/api.py:363]: move it back to debug or remove

File: interactive/dashboard/api.py [hunk 2]
```diff
@@ -1002,3 +1002,15 @@ def post(self, request):
             request.session.create()
         logging.info("saving model options for user: "+request.session.session_key)
         uploaded = request.FILES['file']
+# checks server by trying to connect to default port
+def checkForRabbitMQServer():
```
Comment (K20shores) [inline interactive/dashboard/api.py:1006]: please convert this and all other functions to snake case, which is the standard python naming scheme

Comment (K20shores) [reply interactive/dashboard/api.py:1006]: Actually, delete this. When we try to connect to rabbit that connection will fail and we can just catch that exception

File: interactive/helper.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# from .run_logging import *
+from datetime import datetime
+from pathlib import Path
+import logging
+import hashlib
+
+'''
+For the given path, get the List of all files in the directory tree 
+'''
+def getListOfFiles(dirName):
```
Comment (K20shores) [inline interactive/helper.py:20]: snake case, please

**Comment by K20shores** [comment]
The tests look good to me. Please change the functions that are named in camel case to be named in snake case. Any commented out code (import statements or logging statements) should be deleted.

File: interactive/dashboard/api.py (hunk 2 above)
Comment (jwoody02) [reply interactive/dashboard/api.py]: removed

**Review by K20shores — APPROVED** [review]

File: .github/workflows/run-unit-tests.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the "main" branch
+  push:
+    branches: [ "main", "api-test-cases" ]
```
Comment (mattldawson) [inline .github/workflows/run-unit-tests.yml:9]: you can remove `api-test-cases` now that you know it works

File: interactive/interactive/tools.py [hunk 5]
```diff
@@ -1,12 +1,13 @@
 import os
 import json
-from django.conf import settings
-from django.http import HttpResponse
+# from django.conf import settings
+# from django.http import HttpResponse
```
Comment (mattldawson) [inline interactive/interactive/tools.py:4]: you can just delete these

File: interactive/mechanism/reactions.py [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import json
-from django.conf import settings
+# from django.conf import settings
```
Comment (mattldawson) [inline interactive/mechanism/reactions.py:2]: this can be deleted

File: interactive/helper.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+# helper function getting all files for user session
+from tabnanny import check
+import json
+import os
+import subprocess
+import mimetypes
+# import pandas as pd
+import time
+import shutil
+# from .run_logging import *
```
Comment (mattldawson) [inline interactive/helper.py:11]: you can remove the commented out code

File: interactive/interactive/tools.py [hunk 8]
```diff
@@ -1,12 +1,13 @@
...
 from .unit_dict import *
 from .conversion_dict import *
 from .converter_class import Unit
 
 
+base_dir = '/music-box-interactive/interactive'
```
Comment (mattldawson) [inline interactive/interactive/tools.py:10]: why is this needed? this will only work if the server is running in a docker container built with the Docker file we provide (i.e., with the `music-box-interactive/` folder in the root directory). It seems to me it would be better to use the `settings.BASE_DIR` to allow others (or us) to install differently if needed.

File: interactive/mechanism/reactions.py [hunk 9]
```diff
@@ -9,12 +9,12 @@
 logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
 
 react = "dashboard/static/config/camp_data/reactions.json"
-reactions_default = os.path.join(settings.BASE_DIR, react)
+# reactions_default = os.path.join(settings.BASE_DIR, react)
+reactions_default = os.path.join('/music-box-interactive/interactive', react)
```
Comment (mattldawson) [inline interactive/mechanism/reactions.py:13]: also here - why is it necessary to hard-code the base path?

File: interactive/mechanism/reactions.py [hunk 10]
```diff
@@ -83,7 +83,7 @@ def reaction_menu_names(reactions_path=reactions_default):
 
 # removes a reaction from the mechanism
 def reaction_remove(reaction_index, reactions_path=reactions_default):
-    logging.info('removing reaction ' + str(reaction_index))
+    # logging.info('removing reaction ' + str(reaction_index))
```
Comment (mattldawson) [inline interactive/mechanism/reactions.py:86]: can remove commented out code

File: interactive/mechanism/reactions.py [hunk 11]
```diff
@@ -93,7 +93,7 @@ def reaction_remove(reaction_index, reactions_path=reactions_default):
 
 def remove_reactions_with_species(species, reactions_path=reactions_default,
                                   my_config_path=""):
-    logging.info('removing reactions with species ' + species)
+    # logging.info('removing reactions with species ' + species)
```
Comment (mattldawson) [inline interactive/mechanism/reactions.py:96]: can remove commented out code

File: interactive/mechanism/reactions.py [hunk 12]
```diff
@@ -106,19 +106,19 @@ def remove_reactions_with_species(species, reactions_path=reactions_default,
...
         chem_species = json.loads(f.read())
-    print("chem_species: ", chem_species.keys())
+    # print("chem_species: ", chem_species.keys())
     if 'chemical species' in chem_species.keys():
         if species in chem_species['chemical species']:
-            print("* found species in chemical species")
+            # print("* found species in chemical species")
             del chem_species['chemical species'][species]
-    print("* returning chem_species: ", chem_species)
+    # print("* returning chem_species: ", chem_species)
```
Comment (mattldawson) [inline interactive/mechanism/reactions.py:114]: can remove commented out code

File: interactive/mechanism/species.py [hunk 13]
```diff
@@ -1,18 +1,18 @@
...
 import logging
 import os
 import time
 from interactive.tools import *
 
 logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
 species = "dashboard/static/config/camp_data/species.json"
-species_default = os.path.join(settings.BASE_DIR, species)
+# species_default = os.path.join(settings.BASE_DIR, species)
+species_default = os.path.join('/music-box-interactive/interactive', species)
```
Comment (mattldawson) [inline interactive/mechanism/species.py:11]: also here - why hard-code the base path?

**Review by mattldawson — CHANGES_REQUESTED** [review]
looks good! If you can address the question about the hard-coded base path, go through all the new code and remove commented out lines (I put comments on some, but not all of them), and fix the pep8 stuff, I think it's good-to-go as far as I'm concerned.

**Review by mattldawson — APPROVED** [review]
looks great!
